File: Python-Nitish/BalancedBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalancedBinaryTree(self,root,isbal=True):
        if root:
            l ,isbal = self.isBalancedBinaryTree(root.left,isbal)
            r ,isbal = self.isBalancedBinaryTree(root.right,isbal)

            if r-l not in {-1,0,1}:
                isbal = False 

            return max(l,r)+1,isbal


        return -1,isbal

    def isBalanced(self, root: Optional[TreeNode]) -> bool:
         return self.isBalancedBinaryTree(root)[1]
        

File: java/akash/devsnest-dsa/trees-Binary-Tree/balanced-binary-tree.java
/**
 * Link:- https://devsnest.in/algo-challenges/balanced-binary-tree?tab=question
 * 
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
static int height_of_tree(TreeNode root, int height){
    if(root != null){
        int l = height_of_tree(root.left, height+1);
        int r = height_of_tree(root.right, height+1);

        if(Math.abs(l - r) > 1)
            l = Integer.MAX_VALUE;
        
        return Math.max(l, r);
    }
    return height;
}
static int solve(TreeNode root){
//CODE HERE 
    return height_of_tree(root, 0) == Integer.MAX_VALUE? 0: 1;
}